### PR TITLE
Parcel transformer plugin

### PR DESCRIPTION
**Pull request by stephlow**

### CODE REVIEW

File: packages/parcel-transformer-autometrics/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Transformer } from "@parcel/plugin";
+import { addAutometricsOptions } from "./lib";
+
+export default new Transformer({
+  async transform({ asset }) {
+    const { filePath } = asset;
+    const code = await asset.getCode();
+
+    if (/\.(js|ts)x?$/.test(filePath) && code.includes("autometrics(")) {
```
Comment (arendjr) [inline packages/parcel-transformer-autometrics/src/index.ts:9]: Depending on the formatting, there could be whitespace between `autometrics` and `(`. Might be better to use a regular expression or just check whether it includes `autometrics` (false positives don't have much harm except a little wasted performance, am I right?).

Comment (stephlow) [reply packages/parcel-transformer-autometrics/src/index.ts:9]: Hmmm, that's a good point, I didn't think about that. 

Yeah, false positives basically waste the build performance, nothing else. I'll remove the `(` and we can look into improving the performance if necessary

File: packages/parcel-transformer-autometrics/src/lib.ts [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+              ],
+            ),
+          );
+        } else if (
+          isObjectLiteralExpression(functionOrOptions) &&
+          isFunctionExpression(maybeFunction)
+        ) {
+          const functionName = maybeFunction.name.escapedText.toString();
+
+          let autoMetricsOptions = addObjectPropertyIfMissing(
```
Comment (arendjr) [inline packages/parcel-transformer-autometrics/src/lib.ts:87]: I think we generally write autometrics as one word, right? Then it should also be a lowercase "m" here ;)

**Review by arendjr — APPROVED** [review]
:partying_face:

File: packages/parcel-transformer-autometrics/src/lib.ts (hunk 2 above)
Comment (stephlow) [reply packages/parcel-transformer-autometrics/src/lib.ts]: Good catch! Updated 😄

**Comment by arendjr** [comment]
Looks good to go from my side!

File: .github/workflows/publish_to_npm.yml
Comment (keturiosakys) [inline .github/workflows/publish_to_npm.yml:1]: This should be replaced with a different workflow that publishes the parcel
plugin package independently

File: packages/parcel-transformer-autometrics/package.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  "keywords": [],
+  "main": "dist/index",
+  "scripts": {
+    "dev": "tsc --watch",
+    "build": "tsc",
+    "clean": "rm -rf dist/"
+  },
+  "dependencies": {
+    "@parcel/plugin": "^2.9.0",
+    "typescript": ">=4.9.4"
```
Comment (keturiosakys) [inline packages/parcel-transformer-autometrics/package.json:29]: can TypeScript be a peerDependency or do we need to bundle it for the plugin to
work?

**Review by keturiosakys — APPROVED** [review]
A couple of things about the publishing workflow and bundling typescript but looks good otherwise!

**Review by arendjr — APPROVED** [review]

**Pull request merged**